packages/compass-aggregations/src/modules/pipeline-builder/pipeline-preview-manager.ts

```diff
@@ -30,21 +30,32 @@ export interface PreviewOptions extends AggregateOptions {
   totalDocumentCount?: number;
   sampleSize?: number;
   previewSize?: number;
+  filterOutputStage?: boolean;
+};
+
+const getLastStageOperator = (pipeline: Document[]): string => {
+  return getStageOperator(pipeline[pipeline.length - 1]) ?? ''
 };
 
 export function createPreviewAggregation(
   pipeline: Document[],
   options: Pick<
     PreviewOptions,
-    'sampleSize' | 'previewSize' | 'totalDocumentCount'
+    'sampleSize' | 'previewSize' | 'totalDocumentCount' | 'filterOutputStage'
   > = {}
 ) {
+  const _pipeline = [...pipeline];
+  if (options.filterOutputStage) {
+    const isOutput = isOutputStage(getLastStageOperator(_pipeline));
+    isOutput && _pipeline.splice(_pipeline.length - 1, 1);
+  }
```

Comment:
I don't think this logic belongs in the preview manager, this is only used by text editor view and so should be handled there explicitly

Reply:
I kinda disagree here. Even though its only used in `textEditor`, i think it belongs in preview manager. 

For the output stages, preview is not an option (at least yet). We should instead always filter the last stage if it's an  output stage.

Reply:
I completely agree that previews are not (and can not be) supported by preview manager, but FWIW if we want to make it explicit that preview manager doesn't support out stages we should throw in that case, not implement the logic for removing the stage before getting a preview

Reply:
Implemented [a98e02e](https://github.com/mongodb-js/compass/pull/3681/commits/a98e02ec84825495a1af5fccd92485b45f98ae26)

*Note:* we agreed on throwing error from the preview manager if the last stage of pipeline is an output stage. and the filtering of pipeline to remove last output stage will take place in `PipelineBuilder.getPreviewForPipeline`